### PR TITLE
`Form` : Attachment theming

### PR DESCRIPTION
**Pull request by kaushikrw**
Related to issue: #[apollo/671](https://devtopia.esri.com/runtime/apollo/issues/671)

### Description:

Adds theming support for an `AttachmentsFormElement.

### Summary of changes:

- Added `AttachmentsElementColors` and `AttachmentsElementTypography` to support custom theming for an `AttachmentsFormElement`.
- Added test case to verify the themed properties are displayed correctly.
- Updated `featureforms.api` file to reflect the latest api binary changes.

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [x] link: https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/28/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [x] Yes
  - [ ] No

### CODE REVIEW

File: toolkit/featureforms/api/featureforms.api
Comment (shubham7109) [inline toolkit/featureforms/api/featureforms.api:1]: I'm not sure how to review this, so I assume it looks good.

File: toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/internal/components/attachment/AttachmentTile.kt [hunk 1]
```diff
@@ -369,20 +369,19 @@ private fun Title(
 
 @Composable
 private fun Size(
-    size: Long, modifier:
-    Modifier = Modifier
+    size: Long,
+    modifier: Modifier = Modifier,
+    color : Color = LocalColorScheme.current.attachmentsElementColors.tileTextColor,
```
Comment (shubham7109) [inline toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/internal/components/attachment/AttachmentTile.kt:374]: Formatting: 

```suggestion
    color: Color = LocalColorScheme.current.attachmentsElementColors.tileTextColor,
```

File: toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/theme/FeatureFormDefaults.kt [hunk 2]
```diff
@@ -251,36 +254,71 @@ public object FeatureFormDefaults {
...
+     */
+    @Composable
+    public fun attachmentsElementColors(
+        labelColor : Color = Color.Unspecified,
+        supportingTextColor : Color = Color.Unspecified,
+        outlineColor : Color = MaterialTheme.colorScheme.outline.copy(alpha = 0.6f),
+        containerColor : Color = Color.Unspecified,
+        tileTextColor : Color = MaterialTheme.colorScheme.onSurface,
+        tileBorderColor : Color = MaterialTheme.colorScheme.outline,
+        scrollBarColor : Color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.5f)
```
Comment (shubham7109) [inline toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/theme/FeatureFormDefaults.kt:276]:
```suggestion
        labelColor: Color = Color.Unspecified,
        supportingTextColor: Color = Color.Unspecified,
        outlineColor: Color = MaterialTheme.colorScheme.outline.copy(alpha = 0.6f),
        containerColor: Color = Color.Unspecified,
        tileTextColor: Color = MaterialTheme.colorScheme.onSurface,
        tileBorderColor: Color = MaterialTheme.colorScheme.outline,
        scrollBarColor: Color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.5f)
```

File: toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/theme/FeatureFormTheme.kt [hunk 3]
```diff
@@ -283,6 +286,31 @@ public data class RadioButtonFieldColors internal constructor(
...
+ * @since 200.5.0
+ */
+public data class AttachmentsElementColors internal constructor(
+    public val labelColor: Color,
+    public val supportingTextColor: Color,
+    public val outlineColor: Color,
+    public val containerColor: Color,
+    public val tileTextColor : Color,
+    public val tileBorderColor : Color,
+    public val scrollBarColor : Color,
```
Comment (shubham7109) [inline toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/theme/FeatureFormTheme.kt:311]:
```suggestion
    public val tileTextColor: Color,
    public val tileBorderColor: Color,
    public val scrollBarColor: Color,
```

**Review by shubham7109 — APPROVED** [review]
I have left some minor formatting comments. Changes here work well and looks good to me 👍

File: toolkit/featureforms/api/featureforms.api
Comment (kaushikrw) [reply toolkit/featureforms/api/featureforms.api:1]: I agree its a bit verbose but its all generated so it should be good. Hopefully checks in the future can automate this but.

**Review by sorenoid — APPROVED** [review]

**Pull request merged**